controllers/movies.ts

```diff
@@ -0,0 +1,225 @@
+          minutes = parseInt(movie.Runtime.split(" ")[0]);
+          const episodePerSeason = 10;
+          const seasons = 10;
+          return accumulator + episodePerSeason * seasons * minutes;
+        } else if (movie.Runtime !== "N/A" && movie.Type === "movie") {
+          minutes = parseInt(movie.Runtime.split(" ")[0]);
+          return accumulator + minutes;
+        }
+        return accumulator;
+      }, 0);
+
+      console.log(total);
+      return total;
+    } catch (err) {
+      console.error(err);
+      throw new Error("Calculating total length failed");
+    }
+  }
+
+  async getTotalVotes() {
+    try {
+      const movies = await Movie.find();
+      console.log(movies);
+
+      const total = movies.reduce((accumulator, movie) => {
+        if (movie.imdbVotes !== "N/A") {
+          const votes = parseInt(movie.imdbVotes.replace(",", ""), 10);
+          return accumulator + votes;
+        }
+        return accumulator;
+      }, 0);
+
+      console.log(total);
+      return total;
+    } catch (err) {
+      console.error(err);
+      throw new Error("Calculating total votes failed");
+    }
+  }
+
+  async getImdbUrls() {
+    try {
+      const movies = await Movie.find();
+      console.log(movies);
+      const urlBase = "https://www.imdb.com/";
+      let urls = [];
+      for (const movie of movies) {
+        let movieUrl =
+          urlBase + movie.Title.replace(/\s/g, "") + "/" + movie.imdbID + "/";
+        urls.push(movieUrl);
+      }
+      return urls;
+    } catch (err: any) {
+      console.error(err);
+      throw new Error("Fetching imdbUrls failed");
+    }
+  }
+
+  async getLanguages() {
+    try {
+      const movies = await Movie.find();
+      console.log(movies);
+
+      const languageMappings: { [key: string]: string } = {
+        English: "ENG",
+        Spanish: "ESP",
+        French: "FRA",
+        Russian: "RS",
+      };
+
+      const shortenLanguages: string[] = [];
+
+      for (const movie of movies) {
+        const languages = movie.Language.split(",");
+
+        languages.forEach((language) => {
+          const trimmedLanguage = language.trim();
+          const shortenLanguage = languageMappings[trimmedLanguage] || "ENG";
+          console.log(shortenLanguage);
+          shortenLanguages.push(shortenLanguage);
+        });
+      }
+      return shortenLanguages;
+    } catch (err: any) {
+      console.error(err);
+      throw new Error("Shortening languages failed");
+    }
+  }
+
+  async shortenUrl() {
+    try {
+      const movies = await Movie.find();
+      let shortenedUrls: String[] = [];
+
+      const shortenedURLs = await Promise.all(
+        movies.map(async (movie) => {
+          const randomImageIndex = Math.floor(
+            Math.random() * movie.Images.length
+          );
+          const randomImageURL = await movie.Images[randomImageIndex];
+          if (isValidURL(randomImageURL)) {
```

Comment:
It's okay, but probably you should generate another randomImageUrl if the previous one is invalid